zk/legacy_executor_verifier/legacy_executor_verifier.go

```diff
@@ -264,26 +176,95 @@ func (v *LegacyExecutorVerifier) handleRequest(ctx context.Context, request *Ver
 
 	previousBlock, _ := rawdb.ReadBlockByNumber(tx, blocks[0]-1)
 
-	ok, err := execer.Verify(payload, request, previousBlock.Root())
+	// eager promise will do the work as soon as called in a goroutine, then we can retrieve the result later
+	promise := NewPromise[*VerifierResponse](func() (*VerifierResponse, error) {
+		p := payload
+		r := request
+		root := previousBlock.Root()
```

Comment:
you sure that one won't cause any problems? it captures state from the outside func, not sure if there will be yet another "pointer" issue :-D